fuel-vm/src/interpreter/executors/instruction.rs

```diff
@@ -767,8 +774,13 @@ where
             }
 
             Instruction::CROO(croo) => {
-                self.gas_charge(self.gas_costs().croo)?;
                 let (a, b) = croo.unpack();
+                let contract_id = CheckedMemConstLen::<{ ContractId::LEN }>::new(r!(b))?;
+                let contract_id =
+                    ContractId::from_bytes_ref(contract_id.read(&self.memory));
+                let code_size = contract_size(&self.storage, contract_id)? as Word;
+                let gas_cost = self.gas_costs().croo.resolve(code_size);
+                self.gas_charge(gas_cost)?;
```

Comment:
We need to charge for it in the `code_root`. You can see how it was done in the `code_size`